src/python/pants/backend/python/util_rules/pex_environment.py

```diff
@@ -233,9 +199,94 @@ def first_python_binary() -> Optional[PythonExecutable]:
             python_setup.interpreter_search_paths(pex_relevant_environment)
         ),
         subprocess_environment_dict=subprocess_env_vars.vars,
+        # TODO: This path normalization is duplicated with `engine_initializer.py`. How can we do
+        #  the normalization only once, via the options system?
```

Comment:
For now, use a @memoized_property on the `GlobalOptions` subsystem, similar to:

https://github.com/pantsbuild/pants/blob/1d4dc1ba658ba79f4c554e4eb19c33bf2dddf90f/src/python/pants/backend/shell/lint/shellcheck/subsystem.py#L67-L69

Reply:
It's not that simple - the OptionValueContainervia global_options.options is passed straight into the referenced module and on down. This is an old comment just restored back from here: https://github.com/pantsbuild/pants/commit/8f9138ac04653be5cd51eda8e4332fc8cee0b2f6 You were right back then - there is some figuring to do.